### PR TITLE
Feat: add substitute unstructured object

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/cue/model/value/value.go [hunk 1]
```diff
@@ -82,6 +83,30 @@ func (val *Value) UnmarshalTo(x interface{}) error {
 	return json.Unmarshal(data, x)
 }
 
+// SubstituteInStruct substitute expr in struct lit value
+// nolint:staticcheck
+func (val *Value) SubstituteInStruct(new ast.Expr, key string) error {
```
Comment (yangsoon) [inline pkg/cue/model/value/value.go:88]: The parameter `new` conflicts with the keyword `new`. maybe a different name would be better?

File: pkg/cue/model/value/value.go [hunk 2]
```diff
@@ -82,6 +83,30 @@ func (val *Value) UnmarshalTo(x interface{}) error {
...
+	node := val.CueValue().Syntax(cue.ResolveReferences(true))
+	x, ok := node.(*ast.StructLit)
+	if !ok {
+		return errors.New("value is not a struct lit")
+	}
+	for i := range x.Elts {
```
Comment (yangsoon) [inline pkg/cue/model/value/value.go:94]: Does this mean that only the outermost values can be modified?

Comment (FogDong) [reply pkg/cue/model/value/value.go:94]: Yes I think we can optimize that later since the usage is very limited currently.

**Comment by yangsoon** [comment]
Great Job!

**Review by yangsoon — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**